File: typescript/hamming/hamming.ts
export function compute(left: string, right: string): number {
  if (left.length !== right.length) {
    throw new Error('DNA strands must be of equal length.');
  }

  let distance = 0;
  for (let i = 0; i < left.length; i += 1) {
    if (left[i] !== right[i]) distance += 1;
  }

  return distance;
}

File: c/hamming/hamming.c
#include "hamming.h"

int compute(const char *lhs, const char *rhs)
{
    int dist = 0;

    // iterate over both pointers.
    for (; *lhs && *rhs; lhs += 1, rhs += 1)
    {
        // increase distance if not equal.
        if (*lhs != *rhs) dist += 1;
    }

    // invalid if lhs & rhs are not same length.
    if (*lhs != '\0' || *rhs != '\0') return -1;

    return dist;
}
